### ISSUE TITLE
testing-library/no-debug reporting on non-React Testing Library debug methods

### ISSUE THREAD

**Issue by kirkobyte** [testing-library/eslint-plugin-testing-library#381]
The `testing-library/no-debug` rule is incorrectly reporting a warning on a line in my codebase: `Logger.getInstance().debug({})`, where `getInstance` returns an instance of a `Logger` class, which is unrelated to any of the Testing Libraries.

I'm using version `4.3.0` of the library, with the rule enabled by extending `plugin:testing-library/react`.

**Comment by Belco90** [comment]
Hey @kirkobyte! This is probably related to Aggressive Reporting. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

**Comment by kirkobyte** [comment]
Hey @Belco90, oh right. Yeah, that does seem to be the case. So I suppose the solution would be to move `extends: ['plugin:testing-library/react']` to the overrides section and only apply it to test files?

**Comment by Belco90** [comment]
> Hey @Belco90! Oh right, yeah, that does seem to be the case. So I suppose the solution would be to move `extends: ['plugin:testing-library/react']` to the overrides section and only apply it to test files?

If you are getting errors from `testing-library/no-debug` in non-testing files, that would fix it.

However, if you are getting errors reported in testing files, you would need to use [the shared settings](https://github.com/testing-library/eslint-plugin-testing-library#shared-settings) to narrow `eslint-plugin-testing-library` detection. If this is the case and you are not sure how to handle it, let me know and I can help you with it.

**Comment by kirkobyte** [comment]
Fortunately, the method currently isn't used inside any tests. But if it was, it would also report the rule violation there. For the time being it fixes things for us. But do you think it would be good to change how the detection of the `debug` method works to be more specific? Checking that it's either returned from `render` or called off of `screen`?

**Comment by Belco90** [comment]
> But do you think it would be good to change how the detection of the `debug` method works to be more specific? Checking that it's either returned from `render` or called off of `screen`?

This could definitely be an improvement for the rule (and probably other rules). I don't have the time to address this right now, but I'm leaving the ticket open in case someone else can take care of it.

Thanks for reporting the issue!

**Comment by YozhEzhi** [comment]
Hi! Have same trouble.

`running eslint-plugin-testing-library only against testing files` didn't help me.

**Comment by YozhEzhi** [comment]
For now I'm using this workaround.

```
module.exports = {
    parserOptions: {
        project: './tsconfig.eslint.json',
        ecmaVersion: 2020,
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true,
        },
    },
    plugins: [
        'testing-library',
    ],
    overrides: [
         {
            files: ['**/__tests__/**'],
            extends: ['plugin:testing-library/react'],
        },
    ],
    rules: {
        'create-widget/testing-library/no-debug': OFF, // <---- HERE IS A HACK
    },
};
```

**Comment by Belco90** [comment]
@YozhEzhi thanks for sharing!

Regarding your `overrides` config, I'm not sure if the `files` glob pattern is correct. Could you try with `"**/__tests__/**/*.[jt]s?(x)"` so you indicate the file extension properly?

**Comment by YozhEzhi** [comment]
Sorry, forget to mention that I've tried this:

```
module.exports = {
    parserOptions: {
        project: './tsconfig.eslint.json',
        ecmaVersion: 2020,
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true,
        },
    },
    plugins: [
        'testing-library',
    ],
    overrides: [
         {
            files: ['**/__tests__/**'],
            extends: ['plugin:testing-library/react'],
             rules: {
                    'create-widget/testing-library/no-debug': ERROR, // <---- HERE
             },
        },
    ],
    rules: {
        'create-widget/testing-library/no-debug': OFF,
    },
};
```

And it works as I expected: all codebase has OFF rule, but tests have ERROR for this rule.

**Comment by YozhEzhi** [comment]
Reopen, please.

**Comment by Belco90** [comment]
Hey @YozhEzhi! Not sure what's the expected outcome for this ticket.

I just realized about an issue in your config now:

```js
module.exports = {
    parserOptions: {
        project: './tsconfig.eslint.json',
        ecmaVersion: 2020,
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true,
        },
    },
    plugins: [
        'testing-library',
    ],
    overrides: [
         {
            files: ['**/__tests__/**'],
            extends: ['plugin:testing-library/react'],
             rules: {
                    'create-widget/testing-library/no-debug': ERROR, // <---- HERE
             },
        },
    ],
    rules: {
        'create-widget/testing-library/no-debug': OFF,
    },
};
```

The line where you are pointing the error is setting the wrong rule: it should be `testing-library/no-debug` instead of `create-widget/testing-library/no-debug` (not really sure from where you got that prefix).
